src/lib/provable/test/twisted-curve.unit-test.ts

```diff
@@ -0,0 +1,83 @@
+import { TwistedCurveParams } from '../../../bindings/crypto/elliptic-curve-examples.js';
+import { createCurveTwisted } from '../../../bindings/crypto/elliptic-curve.js';
+import {
+  array,
+  equivalentProvable,
+  map,
+  onlyIf,
+  spec,
+  unit,
+} from '../../testing/equivalent.js';
+import { Random } from '../../testing/random.js';
+import { assert } from '../gadgets/common.js';
+import {
+  Point,
+  CurveTwisted,
+  simpleMapToCurve,
+} from '../gadgets/twisted-curve.js';
+import { foreignField, throwError } from './test-utils.js';
+
+// provable equivalence tests
+const Edwards25519 = createCurveTwisted(TwistedCurveParams.Edwards25519);
+let curves = [Edwards25519];
+
+for (let Curve of curves) {
+  // prepare test inputs
+  let field = foreignField(Curve.Field);
+  let scalar = foreignField(Curve.Scalar);
+
+  // point shape, but with independently random components, which will never form a valid point
+  let badPoint = spec({
```

Comment:
I don't understand why we define `badPoint` as a point which cannot be valid, but then only use it to define a different point which is valid.

Reply:
I am not a fan either, I was copying the format we already had for elliptic curves unit tests

Reply:
if you're not a fan, you can also define the valid point directly

```ts
  let point = spec({
    rng: Random.map(field.rng, (x) => simpleMapToCurve(x, Curve)),
    there: Point.from,
    back: Point.toBigint,
    provable: Point.provable,
  });
  ```